src/blockchain/ain/jellyfish/utxo-provider.service.ts

```diff
@@ -136,16 +137,18 @@ export class UtxoProviderService {
     sizePriority: UtxoSizePriority,
   ): AddressUnspent[] {
     const amountPlusFeeBuffer = amount.plus(Config.blockchain.minFeeBuffer);
-    let total = new BigNumber(0);
-    const neededUnspent: AddressUnspent[] = [];
-    unspent = unspent.sort((a, b) =>
-      sizePriority === UtxoSizePriority.BIG ? this.orderDescending(a, b) : this.orderAscending(a, b),
+    let [neededUnspent, total] = UtxoProviderService.provideUntilAmountPlusFee(
+      unspent,
+      amountPlusFeeBuffer,
+      sizePriority,
     );
-    unspent.forEach((u) => {
-      if (total.gte(amountPlusFeeBuffer)) return;
-      neededUnspent.push(u);
-      total = total.plus(u ? new BigNumber(u.vout.value) : 0);
-    });
+    if (total.lt(amountPlusFeeBuffer) && sizePriority !== UtxoSizePriority.BIG) {
```

Comment:
We only want to retry, if size priority is FITTING, don't we? If yes, I would check `sizePriority === UtxoSizePriority.FITTING`.

Reply:
initially I thought so too, but then changed it because, if we have small und cannot reach the requested amount we should try again with BIG. on the other hand if all UTXOs are currently used, it doesn't matter if we use BIG...

yes will change it